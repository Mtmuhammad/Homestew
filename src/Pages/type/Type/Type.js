import { React, useState, useEffect } from "react";
import SmallPost from "../../../Components/SmallPost/SmallPost";
import Spinner from "../../../Components/Spinner/Spinner";
import client from "../../../http-common";

const Type = ({ type }) => {
  const [recipes, setRecipes] = useState([]);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    setIsLoading(true);
    client
      .get("complexSearch", {
        params: {
          type,
          apiKey: process.env.REACT_APP_API_KEY,
          number: 28,
        },
      })
      .then(function (res) {
        setRecipes(res.data.results);
        setIsLoading(false);
      })
      .catch(function (error) {
        console.log(error);
      });
  }, [type]);
  return (
    <>
      {" "}
      <h3 style={{ marginBottom: "2rem" }} className="tertiary">
        {type} recipes{" "}
        <i
          style={{
            color: "#008080",
            fontSize: "3rem",
          }}
          className="fa-solid fa-utensils"
        ></i>
      </h3>
      <div className="row">
        {isLoading && <Spinner />}
        {recipes && recipes.map((recipe) => (
          <SmallPost
            key={recipe.id}
            id={recipe.id}
            title={recipe.title}
            img={recipe.image}
          />
        ))}
      </div>
    </>
  );
};

export default Type;
