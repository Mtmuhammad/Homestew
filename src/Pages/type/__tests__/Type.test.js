import React from "react";
import { render, screen } from "@testing-library/react";
import Type from "../Type/Type";
import {BrowserRouter } from "react-router-dom"

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
   render(<Type type={"spanish"} />, {wrapper: BrowserRouter});
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<Type type={"spanish"} />, {wrapper: BrowserRouter});
    expect(asFragment()).toMatchSnapshot();
  });
});

test("should render heading", () => {
   render(<Type type={"spanish"} />, {wrapper: BrowserRouter});
   const heading = screen.getByRole("heading", { name: /spanish recipes/i})
   expect(heading).toBeInTheDocument()
   expect(heading).toHaveClass("tertiary")
})