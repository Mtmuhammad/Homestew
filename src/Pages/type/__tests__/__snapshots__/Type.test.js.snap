// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
   
  <h3
    class="tertiary"
    style="margin-bottom: 2rem;"
  >
    spanish recipes 
    <i
      class="fa-solid fa-utensils"
      style="color: rgb(0, 128, 128); font-size: 3rem;"
    />
  </h3>
  <div
    class="row"
  >
    <div
      class="spinner-container"
    >
      <div
        class="loading-spinner"
      />
    </div>
  </div>
</DocumentFragment>
`;
